src/__experimental__/components/checkbox/checkbox-group.component.js

```diff
@@ -7,26 +7,53 @@ import FormField from '../form-field';
 
 const CheckboxGroup = (props) => {
   const {
+    id,
     children,
-    groupName,
     hasError,
     hasWarning,
-    hasInfo
+    hasInfo,
+    onChange,
+    onBlur,
+    value,
+    defaultValue
   } = props;
 
-  const groupLabelId = `${groupName}-label`;
+  const defaultChecked = (defaultValue !== undefined) ? defaultValue : [];
+  const checkboxGroupValue = (value !== undefined) ? value : defaultChecked;
+  const groupLabelId = `${id}-label`;
+  const [checkedValue, setCheckedValue] = useState(checkboxGroupValue);
 
-  const buttons = React.Children.map(children, (child) => {
-    const handleChange = (ev) => {
-      child.props.onChange(ev);
+  const onChangeProp = useCallback((e) => {
+    const checkedCheckboxes = [...checkedValue];
+    const checkedIndex = checkedCheckboxes.indexOf(e.target.value);
+
+    if (checkedIndex === -1) {
+      checkedCheckboxes.push(e.target.value);
+    } else {
+      checkedCheckboxes.splice(checkedIndex, 1);
+    }
+
+    setCheckedValue(checkedCheckboxes);
+
+    e.target = {
+      id,
+      value: checkedCheckboxes
     };
 
+    onChange(e);
+  }, [onChange, checkedValue]);
+
+  const buttons = React.Children.map(children, (child) => {
+    const checked = (checkedValue.indexOf(child.props.value) !== -1);
+
     let childProps = {
-      inputName: groupName,
-      onChange: handleChange
+      name: child.props.name,
```

Comment:
```suggestion
```
This line does not do anything because we're accessing the prop from same child that we're re-applying it to.